### PR TITLE
feat: Add RemovedFromCart event

### PR DESCRIPTION
**Pull request by shayaulman**

### CODE REVIEW

File: src/Http/Controllers/CartItemController.php [hunk 1]
```diff
@@ -29,6 +30,8 @@ public function destroy(OrderItem $item)
     {
         $item->delete();
 
+        event(new RemovedFromCart($item));
```
Comment (Yiddishe-Kop) [inline src/Http/Controllers/CartItemController.php:33]: What if the item is removed by updating the qty to zero?

Isn't it better to trigger this event on the `OrderItem` model `deleted` event?

**Review by Yiddishe-Kop — CHANGES_REQUESTED** [review]

File: src/Http/Controllers/CartItemController.php (hunk 1 above)
Comment (Yiddishe-Kop) [reply src/Http/Controllers/CartItemController.php:33]: The `RemovedFromCart` event should probably only be triggered if the Order is still in the cart status.

Or maybe we can leave that to the listener to check (if it matters)

Comment (shayaulman) [reply src/Http/Controllers/CartItemController.php:33]: > See HandlesCartItems::remove(), that wouldn't trigger this event.

Why not just emitting the event also from there?

> And HandlesCartItems::empty() does a mass deletion [$this->items()->delete()], which wouldn't trigger even a model-event.

I would say that clearing the cart should be considered as a special action and we should probably add a `CartCleared` / `CartEmptied` event for that, I don't think it's justified to clear the items individually (on `HandlesCartItems::empty()`) just to have the event triggered

Comment (Yiddishe-Kop) [reply src/Http/Controllers/CartItemController.php:33]: > Why not just emitting the event also from there?

DRY 👨‍🍳

Besides, it's more logical.

Are there any downsides to that approach?

> `CartEmptied` event 

Totally makes sense to have a separate event for that.

File: src/Events/CartEmptied.php [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+use YiddisheKop\LaravelCommerce\Models\Order;
+use Illuminate\Foundation\Events\Dispatchable;
+
+class CartEmptied
+{
+    use Dispatchable;
+    use SerializesModels;
+
+    public function __construct(
+        public Order $order,
```
Comment (Yiddishe-Kop) [inline src/Events/CartEmptied.php:15]: This package currently supports php 7.4

If we use php 8 features, we'd need to remove php 7.4 support from `composer.json`.

It's a breaking change, so we'd need to tag a new version.

**Pull request merged**